### PR TITLE
pre-commit: lint only changed js files

### PR DESCRIPTION
**Pull request by curioussavage**
:eyeglasses: @schwers or @ajacksified or @nramadas 

this will only lint staged js files on pre commit

### CODE REVIEW

**Comment by curioussavage** [comment]
I could also do this for pushing but as far as I know I would need to use `git diff --name-only <upstream>/master` so we would need a convention for the main repo name.

**Comment by curioussavage** [comment]
I realized client.es6.js was not being linted. last two commits fixes that

**Comment by curioussavage** [comment]
bleh, I didn't realize that what branch I was on when I added the last 3 commits, still a relatively small PR. Unless anyone thinks its an issue I will leave them on this pr

**Comment by curioussavage** [comment]
:eyeglasses: @dwick  for the ads files

File: assets/js/frames.es6.js [hunk 1]
```diff
@@ -164,10 +163,10 @@ const frames = {
     let namespaceProxies = proxies[namespace];
 
     if (namespaceProxies) {
-      namespaceProxies.targets = [].concat(namespaceProxies.targets, target);
+      namespaceProxies.targets = [].concat(namespaceProxies.targets, targets);
```
Comment (curioussavage) [inline assets/js/frames.es6.js]: this looked like a typo..

File: assets/js/frames.es6.js [hunk 2]
```diff
@@ -183,19 +182,18 @@ const frames = {
    * param {Object} [context=this] The context the callback is invoked with.
    * returns {Object} The listener.
    */
-  receiveMessageOnce: function (source, type, callback, context) {
-    return frames.receiveMessage(source, type, function() {
+  receiveMessageOnce(source, type, callback, context) {
+    const listener = frames.receiveMessage(source, type, function() {
```
Comment (curioussavage) [inline assets/js/frames.es6.js]: @dave I saw that listener was not defined but does `receiveMessageOnce` need to return anything?

Comment (curioussavage) [inline assets/js/frames.es6.js]: also, random thought. If this is shared between codebases maybe it could be an npm module?

Comment (schwers) [inline assets/js/frames.es6.js]: :+1: to npm modules idea if it's shared

File: assets/js/client.es6.js [hunk 3]
```diff
@@ -279,7 +276,7 @@ function initialize(bindLinks) {
     refreshMS = Math.max(refreshMS - (1000 * 60), 0);
 
     window.setTimeout(function() {
-      refreshToken(app).then(function(){});
```
Comment (schwers) [inline assets/js/client.es6.js]: :no_bell: We shouldn't need then(...) but this refresh code should just be cleaned up in separate pr

File: assets/js/frames.es6.js (hunk 1 above)
Comment (dwick) [inline assets/js/frames.es6.js]: :+1:

File: assets/js/frames.es6.js (hunk 2 above)
Comment (dwick) [inline assets/js/frames.es6.js]: yeah, i've got a repo for it but havent gotten around to modularizing it.

File: assets/js/frames.es6.js [hunk 4]
```diff
@@ -183,19 +182,18 @@ const frames = {
...
       callback && callback.apply(this, arguments);
-
-      listener.off();
     }, context);
+    listener.off();
```
Comment (dwick) [inline assets/js/frames.es6.js]: this should stay inside the function or else the callback will never be caled

File: assets/js/gtm.es6.js [hunk 5]
```diff
@@ -1,25 +1,23 @@
...
-let jail = document.getElementById('gtm-jail');
-
-let gtm = {
-
-  trigger: function(eventName, payload) {
-    if (payload) {
-      this.set(payload);
+export default {
+  trigger(eventName, payload) {
+    const jailEl = document.getElementById('gtm-jail');
```
Comment (dwick) [inline assets/js/gtm.es6.js]: :no_bell: I think this can be outside of the function.  it's either defined immediately or ever.

**Comment by dwick** [comment]
:nail_care:

**Comment by dwick** [comment]
thanks for fixing this

File: assets/js/client.es6.js [hunk 6]
```diff
@@ -12,16 +12,16 @@ function onError(message, url, line, column) {
...
-    hivemind: window.bootstrap && window.bootstrap.config ? window.bootstrap.config.statsURL : undefined,
+    hivemind: window.bootstrap && window.bootstrap.config ?
+      window.bootstrap.config.statsURL : undefined,
   });
 }
 
 // Register as early as possible
 window.onerror = onError;
 
-import React from 'react';
```
Comment (schwers) [inline assets/js/client.es6.js]: :no_bell: wat :water_buffalo:

File: assets/js/gtm.es6.js (hunk 5 above)
Comment (schwers) [inline assets/js/gtm.es6.js]: Shouldn't we be waiting until onready or etc has happened before looking it up though? `getElementyById` should be fast. Alternatively we could cache this lookup the first time the function is called.

File: assets/js/frames.es6.js (hunk 4 above)
Comment (schwers) [inline assets/js/frames.es6.js]: @dwick been talking to @curioussavage about this. It looks like the current code shouldn't be working since listener isn't in scope when the callback is defined. Should it be this?

```
receiveMessageOnce(source, type, callback, context) {
  const listener = frames.receiveMessage(source, type, function() {
    callback && callback.apply(this, arguments);
    listener.off();
  }, context);

  return listener;
},
```

File: assets/js/gtm.es6.js (hunk 5 above)
Comment (curioussavage) [inline assets/js/gtm.es6.js]: My worry was that since it was being run when the bundle is loaded it could run before that default element renders. Even if that isn't the case now future changes to how the app loads/renders could affect that.

I was using a function that would cache it but ryan pointed out that getElementById should be really fast anyways.

File: assets/js/client.es6.js (hunk 6 above)
Comment (curioussavage) [inline assets/js/client.es6.js]: yep. React isn't used here.

File: assets/js/frames.es6.js [hunk 7]
```diff
@@ -53,11 +52,11 @@ function _addEventListener(type, handler, useCapture) {
   }
 }
 
-function _removeEventListener(type, handler, useCapture) {
+function _removeEventListener(type, handler) {
   if (global.removeEventListener) {
     global.removeEventListener(type, handler);
   } else if (global.detachEvent) {
-    global.attachEvent(`on${type}`, handler);
+    global.detachEvent(`on${type}`, handler);
```
Comment (curioussavage) [inline assets/js/frames.es6.js]: pretty sure this was another typo. but for sanity @dwick ^

**Comment by curioussavage** [comment]
:haircut:  comentz if any please

File: assets/js/frames.es6.js (hunk 2 above)
Comment (schwers) [inline assets/js/frames.es6.js]: @curioussavage The docstring makes me thing `listener` needs to be returned

Comment (schwers) [inline assets/js/frames.es6.js]: yey

**Comment by ajacksified** [comment]
:fish:
